### PR TITLE
try alt formatting rules

### PR DESCRIPTION
**Pull request by V-FEXrt**
Tried some of the easier suggested rules from the other PR. Putting it up for comparison

### CODE REVIEW

File: share/wake/lib/system/environment.wake [hunk 1]
```diff
@@ -13,9 +13,12 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+
 package wake
 
-def uname = prim "uname"
+def uname =
+    prim "uname"
```
Comment (mmjconolly) [inline share/wake/lib/system/environment.wake:20]: We have some polarizing choices for `=` and `\n`, being:
* Zero chars after `=`, then newline
* Up to line-length after `=`, then newline

I can try to come up with some softer in-betweens:

We could have a: 
* Max of (say) 20 chars after `=` (up to regular line length), then newline

or a even a 
* Only allow `[a-zA-Z0-9_.]` after `=`, then newline (ie, only 'simple' looking expressions expression allowed)

File: share/wake/lib/system/job.wake [hunk 2]
```diff
@@ -440,108 +736,189 @@ export def defaultRunner: Runner = fuseRunner
...
+                def pwrite m p d =
+                    prim "write"
+                def json =
+                    JObject
+                    (
+                        "label" :-> JString label,
+                        "command"
+                        :-> command
+                        | map JString
+                        | JArray,
```
Comment (ag-eitilt) [inline share/wake/lib/system/job.wake:790]: I think the human perspective categorizes symbolic operators into distinct categories, which the AST doesn't necessarily capture.

**Review by ag-eitilt — COMMENTED** [review]
I think the previous PR is better *as is* and this one winds up requiring the short-medium-long tristate logic rather than being workable in its current state, but overall I do greatly prefer this.  (And, yes, I know these aren't necessarily all-one or all-the-other.)

File: share/wake/lib/core/list.wake [hunk 3]
```diff
@@ -158,16 +181,22 @@ export def mapFlat (mapFn: a => List b): (list: List a) => List b =
...
+                Nil
+            h, t =
+                # don't wait on f to process tail:
+                def sub =
+                    loop t
+                match (f h)
+                    Some x =
+                        x, sub
+                    None =
+                        sub
```
[the remark this answers is not in the document]
Comment (terpstra) [reply share/wake/lib/core/list.wake:196]: I mean, this code is fine. The formatting is on crack, though.

File: share/wake/lib/system/job.wake (hunk 2 above)
Comment (terpstra) [reply share/wake/lib/system/job.wake:790]: What's super confusing here is that the operator precedence is not respected.
The expression here is `"command" :-> (command | map JSrting | JArray)` and the formatting totally hides that.

File: share/wake/lib/core/json.wake [hunk 4]
```diff
@@ -413,7 +595,12 @@ def mergeValueList (describeCompat: Boolean) (values: List JValue): Result JValu
...
     def compatibilityMessage upper lower error =
         if describeCompat then
-            ("For compatibility, JSON objects should not have duplicate keys. While we try to recover from this where possible, ", lower, error, Nil)
+            (
+                "For compatibility, JSON objects should not have duplicate keys. While we try to recover from this where possible, ",
+                lower,
+                error,
+                Nil
+            )
             | cat
```
[the remark this answers is not in the document]
Comment (bmitc) [reply share/wake/lib/core/json.wake:604]: I am a strong believer that the style of

```
(
    <stuff>
)
```

should be discouraged in a language with ML heritage. It feels very C/C++, C#, Java, and Dart like. I know it's a broken record at this point, haha, but having a literal list syntax would help this a bit, as (as far as I can tell) the usual way to format a multiline list in other MLs is:

```fsharp
[ "For compatibility, ...",
  lower,
  error ]
```

(We should probably verify that, but that's what I feel I've seen, and it's certainly how I currently prefer do it.)

I know I am touching upon a oft visited point, but we are already needing to surround lists with `()` quite often. In wait of the literal list syntax, I'd prefer a list like this to be formatted as

```fsharp
( "For compatibility, ...",
  lower,
  error, )
```

When `[]` comes into play, then the formatter could seemingly "easily" (🙈) reformat this to the above example using `[]`.

**Comment by V-FEXrt** [comment]
The main comments in this PR have been applied to the formatter code. Closing this now for a re-upload with the current state of the world

**Pull request closed without merging**